readthedocs/core/unresolver.py

```diff
@@ -193,16 +237,16 @@ def _normalize_filename(filename):
             filename = "/" + filename
         return filename
 
-    def _match_multiversion_project(self, parent_project, path):
+    def _match_multiversion_project(
+        self, parent_project, path, external_version_slug=None
+    ):
         """
         Try to match a multiversion project.
 
-        If the translation exists, we return a result even if the version doesn't,
-        so the translation is taken as the current project (useful for 404 pages).
+        An exception is raised if we weren't able to find a matching version or language,
+        this exception has the current project (useful for 404 pages).
 
-        :returns: None or a tuple with the current project, version and file.
-         A tuple with only the project means we weren't able to find a version,
-         but the translation was correct.
+        :returns: A tuple with the current project, version and file.
         """
         match = self.multiversion_pattern.match(path)
         if not match:
```

Comment:
Shouldn't this case raise an exception as well? Otherwise, we will be having 3 different outputs here:

* exception
* tuple
* `None`

Reply:
If there isn't a match we try the next code, it nothing matches we raise an exception at the caller `_unresolve_path`.